src/Components/Timer/Timer.tsx

```diff
@@ -1,53 +1,83 @@
 import React, { useState, useEffect } from 'react'
 import './Timer.scss'
+import StopTimerModal from '../StopTimerModal/StopTimerModal'
 
 interface Props {
-  total_seconds: number,
-  stopTimer: (arg0: string) => {}
+  totalSeconds: number,
+  saveWriting: () => void
 }
 
-const Timer: React.FC<Props> = ({ total_seconds, stopTimer }) => {
+const Timer: React.FC<Props> = ({ totalSeconds, saveWriting }) => {
+  const [ showModal, setShowModal ] = useState(false)
+  const [ writingInProgress, setWritingInProgress ] = useState(false)
   const [ elapsedTime, setElapsedTime ] = useState(0)
-  const [ inProgress, toggleInProgress ] = useState(false)
-  let colorClass: string = 'green'
+  const [ modalMessage, setModalMessage ] = useState('')
+  const [ colorClass, setColorClass ] = useState('green')
+
   const timeUpMessage: string = 'Your time is up! Would you like to keep writing or save your work?'
   const stopTimerMessage: string = 'Would you like to keep writing or save your work?'
 
   useEffect(() => {
-    inProgress && setTimeout(() => {setElapsedTime(elapsedTime + 1)}, 1000)
-  }, [elapsedTime])
-
-  useEffect(() => {
-    if (elapsedTime === total_seconds) {
-      toggleInProgress(!inProgress)
+    formatOutlineColor()
+    if (elapsedTime === totalSeconds) {
       stopTimer(timeUpMessage)
-    }
-  })
-
-  useEffect(() => {
-    if (elapsedTime < total_seconds - 30) {
-      colorClass = 'green'
-    } else if (elapsedTime >= total_seconds - 30 && elapsedTime < total_seconds) {
-      colorClass = 'yellow'
-    } else if (elapsedTime >= total_seconds) {
-      colorClass = 'red'
+    } else if (writingInProgress) {
+      incrementTime()
     }
   }, [elapsedTime])
 
+  const incrementTime = () => {
+    setTimeout(() => {setElapsedTime(elapsedTime + 1)}, 1000)
+  }
+
   const toggleTimer = () => {
-    toggleInProgress(!inProgress)
+    if (writingInProgress) {
+      stopTimer(stopTimerMessage)
+      setShowModal(true)
+    } else {
```

Comment:
This code looks clean, however consider weather having that extra else is necessary. you might discover this in the wild that we don't always need to add else statement see my example below. 
` const toggleTimer = () => {
    if (writingInProgress) {
      stopTimer(stopTimerMessage)
      setShowModal(true)
    } 
     setWritingInProgress(true)
     incrementTime()
}
`

Reply:
This function was built to both start and stop the timer. The stopTimer function takes care of the modal and piece of state that holds onto writingInProgress, so it's basically the opposite of of the second part of that function. If I take out the else, would it not just reset the values in state to the opposite of what I wanted? Also, I'll take out the setShowModal from the first part becasue it is already included in the stopTimer function.

Reply:
Good call out Nik! I think it's a great point to refactor though I'm wondering if it would be a bit cleaner with a helper function.

Reply:
I refactored this function and wrote a new function called `startTimer`. This should make this part easier to read.